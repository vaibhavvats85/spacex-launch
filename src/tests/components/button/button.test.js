import { shallow } from 'enzyme';
import Button from '../../../components/button/button';
import React from 'react';

let value, handleClick, wrapper;

beforeEach(() => {
    value = {
        val: '2006',
        id: '2006'
    };
    handleClick = jest.fn();
    wrapper = shallow(<Button value={value} handleClick={handleClick} />);
});

test('should render button corectly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call onClick event', () => {
    wrapper.find('input').prop('onClick')();
    expect(handleClick).toHaveBeenCalled();
});

test('should render label correctly', () => {
    expect(wrapper.find('label').text()).toEqual(value.id);
});