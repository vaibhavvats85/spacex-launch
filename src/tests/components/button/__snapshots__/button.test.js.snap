// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button corectly 1`] = `
<span>
  <input
    defaultChecked={false}
    id="2006"
    onClick={[MockFunction]}
    type="radio"
    value="2006"
  />
  <label
    className="btn"
    htmlFor="2006"
  >
    2006
  </label>
</span>
`;
